### PR TITLE
add account id in the ping request

### PR DESCRIPTION
**Pull request by liyaqin1**
send aws account id of the host from deploy agent to teletraan agent server

### CODE REVIEW

File: deploy-agent/deployd/client/client.py [hunk 1]
```diff
@@ -154,6 +155,9 @@ def _read_host_info(self):
             if not self._stage_type and not self._stage_type_fetched:
                 keys_to_fetch.add(stage_type_key)
 
+            if not self._account_id:
+                keys_to_fetch.add("ec2_metadata.identity-credentials.ec2.info")
```
Comment (ntascii) [inline deploy-agent/deployd/client/client.py:159]: add a helper function in config.py with default value. such as  https://sourcegraph.pinadmin.com/github.com/pinterest/teletraan/-/blob/deploy-agent/deployd/common/config.py?L259

File: deploy-agent/deployd/client/client.py [hunk 2]
```diff
@@ -171,10 +175,16 @@ def _read_host_info(self):
                 self._stage_type = facter_data.get(stage_type_key, None)
                 self._stage_type_fetched = True
 
+            if not self._account_id:
+                ec2_metadata = facter_data.get("ec2_metadata.identity-credentials.ec2.info", None)
+                if ec2_metadata:
+                    info = json.loads(ec2_metadata)
+                    self._account_id = info['AccountId']
```
Comment (ntascii) [inline deploy-agent/deployd/client/client.py:182]: info.get('AccountId') just in case info might not have the key.

**Review by ntascii — COMMENTED** [review]
lgtm, just few nits.

File: deploy-agent/deployd/client/client.py (hunk 2 above)
Comment (liyaqin1) [reply deploy-agent/deployd/client/client.py:182]: sure, make sense, thanks

File: deploy-agent/deployd/client/client.py (hunk 1 above)
Comment (liyaqin1) [reply deploy-agent/deployd/client/client.py:159]: right, make sense

**Comment by liyaqin1** [comment]
Updated devapp’s teletraan agent config to make sure it points to https://teletraan-dev1-agent.pinadmin.com/;
Deployed the related PR https://github.com/pinterest/teletraan/pull/1230 to teletraan agent dev1.
Then tested on the devapp via this command:
```
yaqinli@dev-yaqinli:~/code/teletraan/deploy-agent$ sudo bazel run deployd:deploy-agent -- --use-facter -f /etc/deployagent.conf
INFO: Analyzed target //deployd:deploy-agent (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //deployd:deploy-agent up-to-date:
  bazel-bin/deployd/deploy-agent
INFO: Elapsed time: 0.049s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/deployd/deploy-agent --use-facter -f /etc/deployagent.conf
yaqinli@dev-yaqinli:~/code/teletraan/deploy-agent$
```
According to the log, it looks working well:
```
2023-08-11 00:45:51,039 deployd:120 INFO The deploy agent is starting.
2023-08-11 00:45:51,039 deployd.common.executor:54 DEBUG Executor configs have been updated: PING_INTERVAL=60, TIME_OUT=1800, MAX_RETRY=3
2023-08-11 00:45:51,039 deployd.client.client:55 INFO Use facter to get host info
2023-08-11 00:45:51,039 deployd.common.utils:207 INFO Fetching {'hostname', 'deploy_service_combined', 'ec2_local_ipv4', 'ec2_instance_id'} keys from facter
2023-08-11 00:45:52,753 deployd.common.utils:207 INFO Fetching {'ec2_placement_availability_zone', 'ec2_metadata.identity-credentials.ec2.info', 'deploy_service_stage_type', 'ec2_tags'} keys from facter
2023-08-11 00:45:54,456 deployd.client.client:190 INFO Host information is loaded. Host name: dev-yaqinli, IP: 10.9.215.252, host id: i-0dd4b6ce5062c36f0, agent_version=1.2.40-pinterest-1.0.2, autoscaling_group: None, availability_zone: us-east-1d, stage_type: None, group: ['singer-prod-sidecar', 'envoy-devapp', 'tcollector-prod-sidecar', 'mcrouter-devapp', 'mcrouter-prod-sidecar', 'metrics-agent-prod-sidecar', 'zk_update_monitor-prod-sidecar'], account id: 998131032990
2023-08-11 00:45:54,459 urllib3.connectionpool:1018 DEBUG Starting new HTTPS connection (1): teletraan-dev1-agent.pinadmin.com:443
2023-08-11 00:45:54,533 urllib3.connectionpool:482 DEBUG https://teletraan-dev1-agent.pinadmin.com:443 "POST /v1/system/ping HTTP/1.1" 200 294
2023-08-11 00:45:54,534 deployd.client.client:225 DEBUG PingRequest(hostId=i-0dd4b6ce5062c36f0, hostName=dev-yaqinli, hostIp=10.9.215.252, agentVersion=1.2.40-pinterest-1.0.2, autoscalingGroup=None, availabilityZone=us-east-1d, stageType=None, groups=['singer-prod-sidecar', 'envoy-devapp', 'tcollector-prod-sidecar', 'mcrouter-devapp', 'mcrouter-prod-sidecar', 'metrics-agent-prod-sidecar', 'zk_update_monitor-prod-sidecar'], accountId=998131032990, reports=PingReport(deployId=yo63b11MQ_6Yj8sz5cc8zw, envId=nS2GIXkgQnK4pR9qG5ZpFg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=KIWVKQpJSDu5_CX57jTuWg, envId=inbDfs6KQICpRRfuKQw_bQ, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=3zZuGjXbReqgsgwkAFcQPQ, envId=2XwshUOHR36DGEAH4PzBwg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=sRyuuWtpRKmbbKH56zh49Q, envId=kwUx6aHWQo6R174lT-eawQ, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=4wjThV9dT1SG6dN3vjsevA, envId=typrFRIFR6-lDyaiNjAk9w, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=3jC7i1eNSl6959Nk7bNJOQ, envId=gu2DEmL4Tv2XOqDKFc7Axw, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=2opBs7wjQqmWJGdbBqlY5w, envId=7ib3xSHSR3unnLdN-ViGxw, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=wK80YoJTRaytsW6Ufyczkw, envId=pZwvW9faQTW7vD19ER72Gg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=XgWTORkURUe0HXpkk7dmSw, envId=99xVHd7bSr608HSHqrCGbA, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None),PingReport(deployId=HWT8uaMbTNebUVUhFyGHkw, envId=tLWOA1CsQ4W2Nu_kMFyYVA, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0), errorMessage=None, failCount=0, extraInfo=None, deployAlias=None)) -> PingResponse(opCode=DELETE, deployGoal=DeployGoal(deployId=3zZuGjXbReqgsgwkAFcQPQ, envId=2XwshUOHR36DGEAH4PzBwg, envName=None, stageName=None, deployStage=SERVING_BUILD, build=None, deployAlias=None, agentConfig=None,scriptVariables=None, firstDeploy=False, isDocker=None))
```
And in the database, I see the account_id is recorded:
```
mysql> select * from hosts where host_name="dev-yaqinli";
+-------------+------------------------------------------------------+--------------+---------------+---------------+--------+---------------------+------------+--------------+
| host_name   | group_name                                           | ip           | create_date   | last_update   | state  | host_id             | can_retire | account_id   |
+-------------+------------------------------------------------------+--------------+---------------+---------------+--------+---------------------+------------+--------------+
| dev-yaqinli | envoy-devapp                                         | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | envoy-devapp-production-us-east-1d                   | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | mcrouter-devapp                                      | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | mcrouter-devapp-production-us-east-1d                | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | mcrouter-prod-sidecar                                | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | mcrouter-prod-sidecar-production-us-east-1d          | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | metrics-agent-prod-sidecar                           | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | metrics-agent-prod-sidecar-production-us-east-1d     | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | singer-prod-sidecar                                  | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | singer-prod-sidecar-production-us-east-1d            | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | tcollector-prod-sidecar                              | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | tcollector-prod-sidecar-production-us-east-1d        | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | zk_update_monitor-prod-sidecar                       | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
| dev-yaqinli | zk_update_monitor-prod-sidecar-production-us-east-1d | 10.9.215.252 | 1691714754478 | 1691714754478 | ACTIVE | i-0dd4b6ce5062c36f0 |          0 | 998131032990 |
+-------------+------------------------------------------------------+--------------+---------------+---------------+--------+---------------------+------------+--------------+
14 rows in set (0.00 sec)

mysql> 

```

**Review by ntascii — APPROVED** [review]

**Pull request merged**